bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200Device.java

```diff
@@ -192,22 +194,20 @@ public void listAllServices() {
      * @param actTreeMap
      */
     public void printAllServices(Map<String, KM200ServiceObject> actTreeMap) {
-        if (actTreeMap != null) {
-            for (KM200ServiceObject object : actTreeMap.values()) {
-                if (object != null) {
-                    String val = "", type, valPara = "";
-                    logger.debug("List type: {} service: {}", object.getServiceType(), object.getFullServiceName());
-                    type = object.getServiceType();
-                    if (type == null) {
-                        type = new String();
-                    }
-                    if ("stringValue".equals(type) || "floatValue".equals(type)) {
-                        val = object.getValue().toString();
-                        if (object.getValueParameter() != null) {
-                            if ("stringValue".equals(type)) {
-                                // Type is definitely correct here
-                                @SuppressWarnings("unchecked")
-                                List<String> valParas = (List<String>) object.getValueParameter();
+        for (KM200ServiceObject object : actTreeMap.values()) {
+            String val = "", type, valPara = "";
+            logger.debug("List type: {} service: {}", object.getServiceType(), object.getFullServiceName());
+            type = object.getServiceType();
```

Comment:
I added constants for the different types. I cannot use enum because it cannot be used in switch/case.

Reply:
But enum can be used in switch/case... it is one of the primary benefits of using an enum.

Reply:
but only if I'm using something like:
`public enum DataType {"stringValue", "floatValue")`
It would be in my opinion not really much better. I defined now string constants for this types, like:
` public static final String DATA_TYPE_STRING_VALUE = "stringValue";`
Its ok for you?

Reply:
That is fine for me. An enum type would be better but a constant works as well.